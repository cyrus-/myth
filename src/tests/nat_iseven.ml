let s = Nat_decls.s ^ "

let nat_iseven : nat -> bool |>
  { 0 => True
  | 1 => False
  | 2 => True
  | 3 => False
  } = ?"
