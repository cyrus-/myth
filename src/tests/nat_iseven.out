(* Generated Thu Nov 13 00:35:52 2014 *)
let nat_iseven : nat -> bool =
  let rec f1 (n1:nat) : bool =
    match n1 with
      | O -> True
      | S (n2) -> (match n2 with
                     | O -> False
                     | S (n3) -> f1 n3)
  in
    f1
;;
